### DISCUSSION TITLE
How to set power limits in ROCm 6.2.0

### DISCUSSION THREAD

**Discussion by acaleechurn** [ROCm/ROCm#3557] [Q&A] [answered]
Hello,

I could previously set the power limits (lower) for the MI100s to run a bit cooler with minimal impact on training times using "rocm-smi --setpoweroverdrive 200" but with 6.2.0 this doesn't work anymore. It seems that the commands only accepts values above the advertised power package limit but not lower. I can get it to work again by cleaning the install and run a multi-version setup with the kernel-mode-driver from 6.1.2 but I'm not sure if this is breaking something in 6.2.0.

ROCm 6.2.0

ERROR: GPU[0] : Unable to set Power OverDrive
ERROR: GPU[0] : Value cannot be less than: 290W

ROCm 6.1.2

GPU[0] : Successfully set power to: 200W
GPU[1] : Successfully set power to: 200W

**Comment by rasodu** [comment]
Having the same issue. All of sudden my script have started failing with Rocm 6.2

**Comment by andrerenard** [comment]
I also have this problem, and it's causing issues for my clusters.

**Comment by Beinsezii** [comment]
A few kernel versions ago they changed the amdgpu driver to gate writes to `/sys/class/drm/card1/device/hwmon/power1_cap` based on the card's firmware. Apparently the prior behavior with no floor was unintended. Now if you `cat power1_cap_min` it's only 10% less than `cat power1_cap_default`. I imagine either similar was done to `rocm-smi --setpoweroverdrive` or it itself writes `power1_cap` as the means of control.

**Comment by darren-amd** [comment, marked as answer]
Hi @acaleechurn,

Spoke with the team and we put a firmware update that should fix the issue. It is not yet released but is scheduled to be released with ROCm 6.4. Please let me know if the issue persists, thanks for reporting!

**Comment by acaleechurn** [comment]
Thanks @darren-amd . I will test and confirm once the fix is released.

**Comment by tdavie** [comment]
@darren-amd Hi, just wanted to report that this still appears to be the behaviour as of ROCm 6.4.1. In the case of MI100, `power1_cap_min` == `power1_cap_default` == `power1_cap_max`, i.e. you have no control over power limits.

```
ERROR: GPU[0]	: Unable to set Power OverDrive
ERROR: GPU[0]		: Value cannot be less than: 290W 
```

I'm considering moving to zen-kernel for [their patch](https://github.com/zen-kernel/zen-kernel/issues/344#issuecomment-2000182149) in the meantime.


EDIT: I had forgot to set `amdgpu.ppfeaturemask=0xffffffff` again after a reinstall. Afterwards, this works as expected.
